receiver/tcpcheckreceiver/internal/configtcp/configtcp.go

```diff
@@ -0,0 +1,41 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package configtcp // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/tcpcheckreceiver/internal/configtcp"
+
+import (
+	"net"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+type TCPClientSettings struct {
+	// Endpoint is always required
+	Endpoint string        `mapstructure:"endpoint"`
+	Timeout  time.Duration `mapstructure:"timeout"`
+}
+
+type Client struct {
+	net.Conn
+	Address  string
+	Timeout  time.Duration
+	DialFunc func(network, address string, timeout time.Duration) (net.Conn, error)
+}
+
+// Dial starts a TCP session.
+func (c *Client) Dial(endpoint string, timeout time.Duration) (err error) {
```

Comment:
https://github.com/open-telemetry/opentelemetry-collector/tree/main/config/confignet is probably better all around and less code for you to maintain